yaml-tests/src/test/java/com/apple/foundationdb/relational/yamltests/server/CodeVersionTest.java

```diff
@@ -0,0 +1,106 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.relational.yamltests.server;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CodeVersionTest {
+    private static String randomVersionString(Random r) {
+        if (r.nextBoolean()) {
+            SpecialCodeVersion.SpecialCodeVersionType[] types = SpecialCodeVersion.SpecialCodeVersionType.values();
+            int choice = r.nextInt(types.length);
+            return types[choice].getText();
+        } else {
+            String version = IntStream.generate(() -> r.nextInt(1000))
+                    .limit(4)
+                    .mapToObj(Integer::toString)
+                    .collect(Collectors.joining("."));
+            if (r.nextBoolean()) {
+                version = version + "-SNAPSHOT";
+            }
+            return version;
+        }
+    }
+
+    private static List<String> randomVersionStrings(Random r, int length) {
+        return Stream.generate(() -> randomVersionString(r))
+                .limit(length)
+                .collect(Collectors.toList());
+    }
+
+    private void assertSorted(List<CodeVersion> versions) {
+        int pos = 0;
+        // First, all the min versions
+        while (pos < versions.size() && versions.get(pos).equals(SpecialCodeVersion.min())) {
+            pos++;
+        }
+        // Then all the semantic versions, in order
+        SemanticVersion last = null;
+        while (pos < versions.size() && versions.get(pos) instanceof SemanticVersion) {
+            SemanticVersion current = (SemanticVersion) versions.get(pos);
+            if (last != null) {
+                assertThat(current)
+                        .as("current version %s is out of order in %s", current, versions)
+                        .isGreaterThanOrEqualTo(last);
+            }
+            last = current;
+            pos++;
+        }
+        // Then all the current
+        while (pos < versions.size() && versions.get(pos).equals(SpecialCodeVersion.current())) {
+            pos++;
+        }
+        // Then all the max
+        while (pos < versions.size() && versions.get(pos).equals(SpecialCodeVersion.max())) {
+            pos++;
+        }
+        assertThat(pos)
+                .as("versions %s are out of order starting at position %d", versions, pos)
+                .isEqualTo(versions.size());
+    }
+
+    static Stream<Arguments> sortRandomVersions() {
+        final Random r = new Random(0x5ca1ab13L);
```

Comment:
Should this use `RandomizedTestUtils`? It's possible that's not available in yaml-tests, so maybe not worth bringing in here.